nix/pkgs/mayastor/default.nix

```diff
@@ -29,95 +30,96 @@ let
-              pkgs.lib.hasPrefix (toString (src + "/${allowedPrefix}")) path)
-            allowedPrefixes) src;
+  whitelistSource = src: allowedPrefixes:
+    builtins.filterSource
+      (path: type:
+        pkgs.lib.any
+          (allowedPrefix:
+            pkgs.lib.hasPrefix (toString (src + "/${allowedPrefix}")) path)
+          allowedPrefixes) src;
 
-    mayastor = rustPlatform.buildRustPackage rec {
-      name = "mayastor";
-      cargoSha256 = "0pb5j8wg741zhlv25ccbxyad8n4y87q1b36hq7817l41xjpsh2rh";
-      version = "unstable";
-      src = whitelistSource ../../../. [
-        "Cargo.lock"
-        "Cargo.toml"
-        "csi"
-        "cli"
-        "jsonrpc"
-        "mayastor"
-        "rpc"
-        "spdk-sys"
-        "sysfs"
-        "nvmeadm"
-        # We need to copy git as we use git_version!() in rust, we can also
-        # use use nix to pass the hash if we want to by we should, mosty
-        # likely, go with a proper release version.
-        ".git"
-      ];
 
-      LIBCLANG_PATH = "${pkgs.llvmPackages.libclang}/lib";
+  mayastor = rustPlatform.buildRustPackage rec {
+    name = "mayastor";
+    cargoSha256 = "0pb5j8wg741zhlv25ccbxyad8n4y87q1b36hq7817l41xjpsh2rh";
+    version = "unstable";
+    src = whitelistSource ../../../. [
+      "Cargo.lock"
+      "Cargo.toml"
+      "csi"
+      "cli"
+      "jsonrpc"
+      "mayastor"
+      "rpc"
+      "spdk-sys"
+      "sysfs"
+      "nvmeadm"
+      # We need to copy git as we use git_version!() in rust, we can also
+      # use use nix to pass the hash if we want to by we should, mosty
+      # likely, go with a proper release version.
+      ".git"
+    ];
 
-      # these are required for building the proto files that tonic can't find otherwise.
-      PROTOC = "${pkgs.protobuf}/bin/protoc";
-      PROTOC_INCLUDE = "${pkgs.protobuf}/include";
-      C_INCLUDE_PATH = "${libspdk}/include/spdk";
+    LIBCLANG_PATH = "${pkgs.llvmPackages.libclang}/lib";
 
-      buildInputs = [
-        clang
-        e2fsprogs
-        libaio
-        libiscsi.lib
-        libspdk
-        liburing
-        llvmPackages.libclang
-        numactl
-        openssl
-        pkg-config
-        protobuf
-        utillinux
-        xfsprogs
-        utillinux.dev
-      ];
+    # these are required for building the proto files that tonic can't find otherwise.
+    PROTOC = "${pkgs.protobuf}/bin/protoc";
+    PROTOC_INCLUDE = "${pkgs.protobuf}/include";
+    C_INCLUDE_PATH = "${libspdk}/include/spdk";
 
-      buildType = "debug";
-      verifyCargoDeps = false;
+    buildInputs = [
+      clang
+      e2fsprogs
+      libaio
+      libiscsi.lib
+      libspdk
+      liburing
+      llvmPackages.libclang
+      numactl
+      openssl
+      pkg-config
+      protobuf
+      utillinux
+      xfsprogs
+      utillinux.dev
+    ];
 
-      doCheck = false;
-      meta = { platforms = stdenv.lib.platforms.linux; };
-    };
+    buildType = if release then "debug" else "release";
```

Comment:
looks like the condition is inverted.

Reply:
Yes Jan, thanks, @blaisedias also mentioned this to mee this morning. After considering for a while, I think I want to redo this file in a bit more,  but later. 

I think its cleaner to have the docker images themselves split out into separate files. The processes would remain the same, but the logic is elsewhere. Mayastor/default.nix would then only have one artefact instead of multiple things.